### PR TITLE
Rename FrequentSequenceFinder class in spam module to Spam

### PR DESCRIPTION
**Pull request by JSAbrahams**
Rename the confusing `allPath` variable and its derivatives to `sequence` and its derivates:
* `sequence` and its derivatives are used to denote sequences, which more closely coincides with literature
* `patterns` and its derivates are used to denote the output of the algorithm, to more closely coincide with the name of the pipeline

Also rename `FrequentSequenceFinder` to `PrefixSpan` to make clear it uses the PrefixSpan algorithm.
* Make the class internal as it is not used elsewhere.

### CODE REVIEW

File: modules/pipeline/spam-pattern-detector/src/main/kotlin/org/cafejojo/schaapi/pipeline/patterndetector/spam/SPAM.kt [hunk 1]
```diff
@@ -7,44 +7,44 @@ import org.cafejojo.schaapi.models.PathUtil
...
+ * @property minimumSupport the minimum amount of times a node must appear in [sequences] for it to be considered a
  * frequent node.
- * @property comparator the comparator used to determine whether two [Node]s are equal
+ * @property nodeComparator the nodeComparator used to determine whether two [Node]s are equal
  */
-class FrequentSequenceFinder<N : Node>(
-    private val allPaths: Collection<List<N>>,
-    private val minimumCount: Int,
-    private val comparator: GeneralizedNodeComparator<N>
+internal class SPAM<N : Node>(
```
Comment (FWDekker) [inline modules/pipeline/spam-pattern-detector/src/main/kotlin/org/cafejojo/schaapi/pipeline/patterndetector/spam/SPAM.kt]: `SPAM` or `Spam`?

For consistency, I'd go for `Spam`.

**Comment by gandreadis** [comment]
@JSAbrahams About the new PR title: You're not changing the PatternDetector class name, right? Only the FSF

File: modules/pipeline/spam-pattern-detector/src/main/kotlin/org/cafejojo/schaapi/pipeline/patterndetector/spam/SPAM.kt (hunk 1 above)
Comment (gandreadis) [reply modules/pipeline/spam-pattern-detector/src/main/kotlin/org/cafejojo/schaapi/pipeline/patterndetector/spam/SPAM.kt]: +1

**Comment by JSAbrahams** [comment]
@gandreadis Yes you're right, I changed it.

**Pull request merged**